x/distribution/keeper/keeper.go

```diff
@@ -79,9 +79,12 @@ func (k Keeper) WithdrawValidatorCommission(ctx sdk.Context, valAddr sdk.ValAddr
 	}
 
 	coins, remainder := commission.TruncateDecimal()
-
-	// leave remainder to withdraw later
-	k.SetValidatorAccumulatedCommission(ctx, valAddr, remainder)
+	if remainder.IsZero() {
+		k.SetValidatorAccumulatedCommission(ctx, valAddr, types.InitialValidatorAccumulatedCommission())
+	} else {
+		// leave remainder to withdraw later
+		k.SetValidatorAccumulatedCommission(ctx, valAddr, remainder)
+	}
```

Comment:
I think we should move this introduced if-else statement _into_ `SetValidatorAccumulatedCommission` this seems like a general precaution.
 
(here are the other instances of where it's called) 
```
x/distribution/keeper/querier_test.go:140:	keeper.SetValidatorAccumulatedCommission(ctx, valOpAddr1, commission)
x/distribution/keeper/validator.go:18:	k.SetValidatorAccumulatedCommission(ctx, val.GetOperator(), types.InitialValidatorAccumulatedCommission())
x/distribution/keeper/keeper.go:84:	k.SetValidatorAccumulatedCommission(ctx, valAddr, remainder)
x/distribution/keeper/keeper_test.go:44:	keeper.SetValidatorAccumulatedCommission(ctx, valOpAddr3, valCommission)
x/distribution/keeper/allocation.go:93:	k.SetValidatorAccumulatedCommission(ctx, val.GetOperator(), currentCommission)
x/distribution/genesis.go:23:		keeper.SetValidatorAccumulatedCommission(ctx, acc.ValidatorAddress, acc.Accumulated)
```

Reply:
Updated `SetValidatorAccumulatedCommission` -- although on second glance it seems weird to have this business logic in a store writer, but I guess it's fine either way.

Reply:
I think it's OK for now